sound/soc/sof/intel/hda.c

```diff
@@ -591,11 +591,7 @@ int hda_dsp_probe(struct snd_sof_dev *sdev)
 	ret = hda_dsp_stream_init(sdev);
 	if (ret < 0) {
 		dev_err(&pci->dev, "error: failed to init streams\n");
-		/*
-		 * not all errors are due to memory issues, but trying
-		 * to free everything does not harm
-		 */
-		goto err;
+		goto free_streams;
```

Comment:
this is the first idea flashing into my mind.  To free memory at different stage is complex. I like the style of hda_dsp_stream_free. @plbossart which one do you prefer?

Reply:
Instead of fixing the code, I decided to fix the comment... See https://github.com/thesofproject/linux/pull/274/commits/3a420e92d2bfbf3a0c279caef61b79494b90261c